### PR TITLE
Feature: Support namespace-scoped informer

### PR DESCRIPTION
**Pull request by yue9944882**

### CODE REVIEW

File: spring/src/test/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerCreatorTest.java [hunk 1]
```diff
@@ -28,12 +28,7 @@
 import io.kubernetes.client.informer.SharedInformerFactory;
 import io.kubernetes.client.informer.cache.Lister;
 import io.kubernetes.client.openapi.ApiClient;
-import io.kubernetes.client.openapi.models.V1ConfigMap;
-import io.kubernetes.client.openapi.models.V1ConfigMapList;
-import io.kubernetes.client.openapi.models.V1ListMeta;
-import io.kubernetes.client.openapi.models.V1ObjectMeta;
-import io.kubernetes.client.openapi.models.V1Pod;
-import io.kubernetes.client.openapi.models.V1PodList;
+import io.kubernetes.client.openapi.models.*;
```
Comment (brendandburns) [inline spring/src/test/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerCreatorTest.java:31]: no wildcard imports, please.

File: util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java [hunk 2]
```diff
@@ -19,10 +19,7 @@
 import io.kubernetes.client.openapi.ApiClient;
 import io.kubernetes.client.openapi.ApiException;
 import io.kubernetes.client.openapi.Configuration;
-import io.kubernetes.client.util.CallGenerator;
-import io.kubernetes.client.util.CallGeneratorParams;
-import io.kubernetes.client.util.Watch;
-import io.kubernetes.client.util.Watchable;
+import io.kubernetes.client.util.*;
```
Comment (brendandburns) [inline util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java:22]: here too wrt wildcards.

**Comment by brendandburns** [comment]
two really minor nits. We should set up an auto-formatter to organize imports so that we don't have to worry about this.

otherwise LGTM

File: spring/src/test/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerCreatorTest.java (hunk 1 above)
Comment (yue9944882) [reply spring/src/test/java/io/kubernetes/client/spring/extended/controller/KubernetesInformerCreatorTest.java]: fixed

File: util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java (hunk 2 above)
Comment (yue9944882) [reply util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java:22]: fixed

Comment (brendandburns) [reply util/src/main/java/io/kubernetes/client/informer/SharedInformerFactory.java:22]: I don't see the fix :)

File: util/src/test/java/io/kubernetes/client/informer/SharedInformerFactoryTest.java [hunk 3]
```diff
@@ -12,36 +12,87 @@
 */
 package io.kubernetes.client.informer;
 
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+import static org.mockito.Mockito.*;
```
Comment (brendandburns) [inline util/src/test/java/io/kubernetes/client/informer/SharedInformerFactoryTest.java:17]: please remove wildcards (and below)

**Comment by brendandburns** [comment]
Still seeing some wildcard imports. (sorry to be so nit-picky, but I want to keep us to a code style) we should really figure out how to block them in the validation, but it seems like spotless doesn't support this out of the box.

File: util/src/test/java/io/kubernetes/client/informer/SharedInformerFactoryTest.java (hunk 3 above)
Comment (yue9944882) [reply util/src/test/java/io/kubernetes/client/informer/SharedInformerFactoryTest.java]: fixed

**Pull request merged**